sympy/solvers/solveset.py

```diff
@@ -210,6 +261,61 @@ def _invert_complex(f, g_ys, symbol):
                                                log(Abs(g_y))), S.Integers)
                                for g_y in g_ys])
             return _invert_complex(f.args[0], exp_invs, symbol)
+
+    if isinstance(f, sin) or isinstance(f, csc):
```

Comment:
We don't need inverses of trigonometric functions in the complex domain, they can be very easily computed using the exponential definition of trigonometric functions [#1]. Also, sympy inverse functions like `atan` are only defined on real intervals (see `atan` docstring).

[#1]: https://en.wikipedia.org/wiki/Trigonometric_functions#Relationship_to_exponential_function_and_complex_numbers